R/parse-preref.R

```diff
@@ -15,20 +15,24 @@ parse.preref <- function(lines) {
   ## Compress the escaped delimeters.
   elements <- str_replace_all(elements, fixed("@@"), "@")
 
-  parsed <- parse_elements(elements[-1])
-
-  if (elements[[1]] != "") {
-    check_rd(NULL, elements[[1]])
-    parsed$introduction <- str_trim(elements[[1]])
-  }
-  parsed
+  parse_elements(elements)
 }
 
 # Sequence that distinguishes roxygen comment from normal comment.
 LINE.DELIMITER <- '\\s*#+\' ?'
 
 parse_elements <- function(elements) {
-  pieces <- str_split_fixed(elements, "[[:space:]]+", 2)
+
+  pieces <- str_split_fixed(elements[-1], "[[:space:]]+", 2)
+  desc <- process_description(str_trim(elements[1]), pieces)
+  pieces <- rbind(desc, pieces, deparse.level = 0)
+
+  ## Merge multiple @details tags, one might have come from the intro
+  didx <- which(pieces[,1] == "details")
```

Comment:
No, the comment is wrong there, actually. It would be slightly more difficult to handle it there. Now `other_pieces` is an input only in `process_description`, but not output. Handling it there would mean that you need to output/change it (remove or update the existing `details` field), and the output value of the function gets more messy. That's why I moved it up here.